bin/build.sh

```diff
@@ -1,73 +1,81 @@
 
     # Replace wca{...} and wcadoc{...} with absolute WCA URLs (the correct URLs only get inserted in production)
     sed -Ei "s#wca\{([^}]*)\}#$wca_url\1#g" "$file"
     sed -Ei "s#wcadoc\{([^}]*)\}#$wca_docs_url\1#g" "$file"
     # Replace {logo} with the path to the WCA logo in /assets
     sed -Ei "s#\{logo\}#$logo#g" "$file"
 
-    # This creates the custom header for each edudoc and saves it next to the doc that will use it
-    if [ $1 = "edudoc" ]; then
-      header_html="${file%.md}-header.html"
-    
+    # This creates the custom header for each edudoc and prepends it to each file
+    if [ "$1" = "edudoc" ]; then
       # Set the document title, path to the logo and the date
-      sed -E "s#DOCUMENT_TITLE#$document_title#g" "$edudoc_header" |
-      sed -E "s#WCA_LOGO_PATH#$logo#g" |
-      sed -E "s#DATE#$compile_date#g" > "$header_html"
+      sed -E "s#\{document_title\}#$document_title#g" "$edudoc_header" |
+      sed -E "s#\{wca_logo_path\}#$logo#g" |
+      sed -E "s#\{date\}#$compile_date#g" |
+      echo -e "$(cat /dev/stdin)\n\n$(cat $file)" > "$file"
     fi
 
-    # Markdown -> HTML
-    pandoc -s --from markdown --to html5 --css=$2 --metadata pagetitle="$document_title" "$file" -o "$html_name"
+    # Markdown => HTML + remove unsupported styles generated by pandoc
+    pandoc -s --from markdown --to html5 --css "$main_stylesheet" --css "$custom_stylesheet" --metadata pagetitle="$document_title" "$file" |
+    sed -E "s#gap: min\(4vw, 1\.5em\);##" |
+    sed -E "s#overflow-x: auto;##" > "$html_path"
 
     # HTML -> PDF
-    if [ $1 = "documents" ]; then
-      wkhtmltopdf --encoding 'utf-8' -T 15mm -B 15mm -R 15mm -L 15mm --quiet "$html_name" "$pdf_name"
-    elif [ $1 = "edudoc" ]; then
-      wkhtmltopdf --encoding 'utf-8' -T 15mm -B 15mm -R 15mm -L 15mm --header-html "$header_html" --footer-center "[page]" --quiet "$html_name" "$pdf_name"
+    if [ "$1" = "documents" ]; then
+      # --presentational-hints makes things like the start attribute of ol elements work,
+      # which is required for adding more content between two consecutive ordered list items
+      weasyprint --presentational-hints --encoding 'utf-8' "$html_path" "$pdf_path"
+    elif [ "$1" = "edudoc" ]; then
+      weasyprint --presentational-hints --encoding 'utf-8' "$html_path" "$pdf_path"
     fi
 
-    # Deliberately NOT using the pipe | operator between pandoc and wkhtml because wkhtml tries to resolve image links
-    # based on the location of its input file. If piping is used, wkhtml assumes /tmp as location,
-    # and links like "./dummy-image.png" become "/tmp/dummy-image.png" instead of "edudoc/$project/dummy-image.png"
+    # Remove the Markdown and HTML files
+    rm "$file" "$html_path"
   done
 }
 
-# Remove potentially cached PDFs from the last build run and make an empty build folder
-rm -rf build
-mkdir build
+# Create the build directory if it didn't exist in the first place
+mkdir -p build
```

Comment:
In line with my other comment above (feel free to answer only one of them): Why did you move this `rm` step that is clearly present in the left side of the diff into the conversion method body?

Reply:
Answered above